pepper-apis/src/main/java/org/broadinstitute/ddp/service/ActivityInstanceService.java

```diff
@@ -457,6 +447,27 @@ public void countQuestionsAndAnswers(Handle handle, String userGuid, String oper
                 summary.setNumQuestionsAnswered(questionAndAnswerCounts.getRight());
             }
         }
+
+        return instanceResponses;
+    }
+
+    private FormActivityDef getDefinitionForSummary(Handle handle, ActivityDefStore activityDefStore,
+                                                    String studyGuid, ActivityInstanceSummary summary) {
+        String activityCode = summary.getActivityCode();
+        String versionTag = summary.getVersionTag();
+        long versionId = summary.getVersionId();
+        long revisionStart = summary.getRevisionStart();
+
+        FormActivityDef formActivityDef = activityDefStore.getActivityDef(studyGuid, activityCode, versionTag);
+        if (formActivityDef == null) {
```

Comment:
Same block of code exists in `ActivityInstanceDao.java` - could be extracted to method and reused in both places:
```
                       Optional<ActivityDto> activityDto = handle.attach(JdbiActivity.class)
                                .findActivityByStudyGuidAndCode(studyGuid, activityCode);
                        formActivityDef = handle.attach(FormActivityDao.class).findDefByDtoAndVersion(
                                activityDto.get(), versionTag, versionId, revisionStart);
                        activityDefStore.setActivityDef(studyGuid, activityCode, versionTag, formActivityDef);
 ```

Reply:
I think I'll keep it for now.